src/prompt/variable.rs

```diff
@@ -49,8 +237,65 @@ pub struct Uuid;
 
 impl VariableInput for Uuid {
     fn type_name(&self) -> &str { "uuid" }
-    fn parse(&self, input: &str) -> Result<Value, Error> {
-        Ok(Value::Uuid(input.parse().map_err(no_pos_err)?))
+    fn parse<'a>(&self, input: &'a str, _flags: InputFlags) -> ParseResult<'a> {
+        fn no_dashes(i: &str) -> ParseResult<&str, uuid::Uuid> {
+            map_res(
+                nom::bytes::streaming::take_while_m_n(
+                    32usize,
+                    32usize,
+                    char::is_alphanumeric,
+                ),
+                |s| uuid::Uuid::from_str(s).context("Cannot parse to UUID")
+            )(i)
+        }
+
+        fn dashes(i: &str) -> ParseResult<&str, uuid::Uuid> {
+            map_res(
+                tuple((
+                    nom::bytes::streaming::take_while_m_n(
+                        8,
+                        8,
+                        char::is_alphanumeric,
+                    ),
+                    char('-'),
+                    nom::bytes::streaming::take_while_m_n(
+                        4,
+                        4,
+                        char::is_alphanumeric,
+                    ),
+                    char('-'),
+                    nom::bytes::streaming::take_while_m_n(
+                        4,
+                        4,
+                        char::is_alphanumeric,
+                    ),
+                    char('-'),
+                    nom::bytes::streaming::take_while_m_n(
+                        4,
+                        4,
+                        char::is_alphanumeric,
+                    ),
+                    char('-'),
+                    nom::bytes::streaming::take_while_m_n(
+                        12,
+                        12,
+                        char::is_alphanumeric,
+                    ),
+                )),
+                |s| uuid::Uuid::from_str(&[s.0, s.2, s.4, s.6, s.8].join("")).context("UUID format is malformed")
+            )(i)
+        }
```

Comment:
I would just grab 36 characters and pass it to `Uuid::from_str` instead of trying to parse it out directly

Reply:
should we allow this kind of input?
```
33d3ca20-149c-4a49-b8ac-a5a8ff9cc6b3 // OK
33d3ca20149c4a49-b8ac-a5a8ff9cc6b3-- // OK
----33d3ca20149c4a49b8aca5a8ff9cc6b3 // OK
```

Reply:
or wait~ does `Uuid::from_str` validate all that? iirc it doesn't allow `-`

Reply:
oh it does allow, will fix